src/iarc7_motion/iarc_tasks/hold_position_task.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python
+
+import math
+import rospy
+import tf2_ros
+import tf2_geometry_msgs
+import threading
+
+from geometry_msgs.msg import TwistStamped
+from geometry_msgs.msg import Point
+from geometry_msgs.msg import PointStamped
+from nav_msgs.msg import Odometry
+
+from iarc7_msgs.msg import OdometryArray
+
+from .abstract_task import AbstractTask
+from iarc_tasks.task_states import (TaskRunning,
+                                    TaskDone,
+                                    TaskCanceled,
+                                    TaskAborted,
+                                    TaskFailed)
+from iarc_tasks.task_commands import (VelocityCommand,
+                                      ArmCommand,
+                                      NopCommand)
+
+from height_checker import HeightChecker
+
+class HoldPostionTaskStates:
+    init = 0
+    waiting = 1
+    holding = 2
+
+class HoldPositionTask(AbstractTask):
+
+    def __init__(self, task_request):
+        self._hold_current_position = task_request.hold_current_position
+
+        if not self._hold_current_position:
+            self._x_position = task_request.x_position
+            self._y_position = task_request.y_position
+            self._z_position = task_request.z_position
+        else: 
+            self._x_position = None
+            self._y_position = None
+            self._z_position = None
+
+        self._drone_odometry = None
+        self._canceled = False
+
+        self._lock = threading.RLock()
+        self._height_checker = HeightChecker()
+
+        self._current_velocity_sub = rospy.Subscriber(
+            '/odometry/filtered', Odometry,
+            self._current_velocity_callback)
+
+        if not self._hold_current_position:
```

Comment:
Can we move this up to around line 41 so we don't have to have this if statement twice?